### PR TITLE
Provide a callback in `GitPushDialog` that executes when push is complete

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GitPushDialog.java [hunk 1]
```diff
@@ -64,6 +80,30 @@ private void push() {
...
+    executeAfterRunningPrePushHandlers(new Task.Backgroundable(myProject, title, /* canBeCancelled */ true) {
+      @Override
+      public void run(ProgressIndicator indicator) {
+        myController.push(forcePush);
+      }
+
+      @Override
+      @UIEffect
+      public void onSuccess() {
+        doInUIThreadWhenReady.run();
```
Comment (mkondratek) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GitPushDialog.java:102]: btw I think we should rather rename it from `doInUIThreadWhenReady` to `doInUiThreadWhenReady` someday everywhere

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToRemote.java [hunk 2]
```diff
@@ -208,9 +214,12 @@ private boolean handleDivergedFromAndNewerThanRemote(IManagedBranchSnapshot gitM
 
     switch (forcePushApprovalDialogBuilder.show(project)) {
       case YES :
-        // TODO (#1424): git push continues in background
-        new GitPushDialog(project, gitRepository, GitPushSource.create(localBranch), /* isForcePushRequired */ true).show();
-        return true;
+        Runnable doInUIThreadWhenReady = () -> graphTable.queueRepositoryUpdateAndModelRefresh(traverseNextEntry);
+        new GitPushDialog(project, gitRepository, GitPushSource.create(localBranch), /* isForcePushRequired */ true,
+            doInUIThreadWhenReady).show();
+        // The ongoing traverse is now a responsibility of the freshly-queued backgroundable;
```
Comment (mkondratek) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToRemote.java:220]: copy paste error detected 🔴  `backgroundable` - not precise

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToRemote.java [hunk 3]
```diff
@@ -183,9 +186,12 @@ private boolean handleAheadOfRemote(IManagedBranchSnapshot gitMacheteBranch, Git
 
     switch (pushApprovalDialogBuilder.show(project)) {
       case YES :
-        // TODO (#1424): git push continues in background
-        new GitPushDialog(project, gitRepository, GitPushSource.create(localBranch), /* isForcePushRequired */ false).show();
-        return true;
+        Runnable doInUIThreadWhenReady = () -> graphTable.queueRepositoryUpdateAndModelRefresh(traverseNextEntry);
+        new GitPushDialog(project, gitRepository, GitPushSource.create(localBranch), /* isForcePushRequired */ false,
+            doInUIThreadWhenReady).show();
+        // The ongoing traverse is now a responsibility of the freshly-queued backgroundable;
```
Comment (mkondratek) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToRemote.java:192]: copy paste error detected 🔴 as above ☝️

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToRemote.java [hunk 4]
```diff
@@ -158,9 +158,12 @@ private boolean handleUntracked(IManagedBranchSnapshot gitManagedBranch, GitLoca
...
```
Comment (mkondratek) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToRemote.java:164]: copy paste error detected 🔴 as above ☝️

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GitPushDialog.java [hunk 5]
```diff
@@ -64,6 +80,30 @@ private void push() {
...
   }
 
+  /**
+   * Overridden to provide a way to perform an action once the push is completed successfully.
+   * Compare to the {@link com.intellij.dvcs.push.ui.VcsPushDialog#push(boolean)}
+   * which doesn't implement {@code onSuccess} callback.
+   */
+  @Override
+  @UIEffect
+  public void push(boolean forcePush) {
```
Comment (mkondratek) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GitPushDialog.java:90]: nice 🚀

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GitPushDialog.java [hunk 6]
```diff
@@ -64,6 +80,30 @@ private void push() {
...
+
+    String title = DvcsBundle.message("push.process.pushing");
```
Comment (mkondratek) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GitPushDialog.java:92]: nit: it's dangerous to rely on their classes and bundles here. they can change DvcsBundle to GitBundle someday, the key, or the value

**Review by mkondratek — CHANGES_REQUESTED** [review]

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToRemote.java (hunk 2 above)
Comment (mkondratek) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToRemote.java:220]: oh I see the next PR is removing these comments so feel free to ignore me on this one

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GitPushDialog.java (hunk 6 above)
Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GitPushDialog.java:92]: Right... replaced with an identical string from our bundle

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GitPushDialog.java (hunk 1 above)
Comment (mkondratek) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GitPushDialog.java:103]: lol, ok

**Review by mkondratek — APPROVED** [review]

**Pull request merged**